src/coreclr/jit/emitarm64.cpp

```diff
@@ -13597,6 +13597,44 @@ regNumber emitter::emitInsTernary(instruction ins, emitAttr attr, GenTree* dst,
         // src2 can only be a reg
         assert(!src2->isContained());
     }
+    else if ((src1->OperIs(GT_MUL) && src1->isContained()) || (src2->OperIs(GT_MUL) && src2->isContained()))
+    {
+        assert(ins == INS_add);
+
+        GenTree* mul;
+        GenTree* c;
+        if (src1->OperIs(GT_MUL))
+        {
+            mul = src1;
+            c   = src2;
+        }
+        else
+        {
+            mul = src2;
+            c   = src1;
+        }
+
+        GenTree* a = mul->gtGetOp1();
+        GenTree* b = mul->gtGetOp2();
+
+        assert(varTypeIsIntegral(mul) && !mul->gtOverflow());
+
+        bool msub = false;
+        if (a->OperIs(GT_NEG) && a->isContained())
+        {
+            a    = a->gtGetOp1();
+            msub = true;
+        }
+        if (b->OperIs(GT_NEG) && b->isContained())
+        {
+            b    = b->gtGetOp1();
+            msub = !msub; // it's either "a * -b" or "-a * -b" which is the same as "a * b"
```

Comment:
> // it's either "a * -b" or "-a * -b" which is the same as "a * b"

This comment is little misleading...I can see how `-a * -b` is `a * b`, but then we will still use `msub` so probably we should not confuse by saying "same as a * b"

Reply:
> but then we will still use msub

actually for `-a * -b` this code will emit `madd` (msub = !msub will return false)

Reply:
ah...I misread this `if` as `else if`. Never mind.